platform-docs/docs/intro.md

```diff
@@ -30,13 +30,39 @@ To build a block editor, you need to install the following dependencies:
 
 We're going to be using JSX to write our UI and components. So one of the first steps we need to do is to configure our build tooling, By default vite supports JSX and and outputs the result as a React pragma. The Block editor uses React so there's no need to configure anything here but if you're using a different bundler/build tool, make sure the JSX transpilation is setup properly.
 
+## Side note: `process.env.IS_GUTENBERG_PLUGIN`
+
+The Gutenberg block editor reads from the `process.env.IS_GUTENBERG_PLUGIN` variable. It
+is used to distinguish the code that is part of the Gutenberg block editor
+[plugin](https://wordpress.org/plugins/gutenberg/) and the code that is part of
+WordPress core. 
+
+For our purposes, as we're building a standalone block editor that is completely
+separate from WordPress, we want to set this variable to `false`. 
+
+When using Vite, we can do it by creating a `vite.config.js` file:
+
+```js
+// vite.config.js
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  define: {
+    "process.env.IS_GUTENBERG_PLUGIN": JSON.stringify(false),
```

Comment:
> My understanding is that we cannot use import.meta unless we ship our scripts as modules. 

FWIW, I think we already publish our scripts as modules. So is the idea to use `process.env` for the CJS build and `import.meta` for the modules build? That might be the best approach no?

Reply:
IDK why Gutenberg uses `process.env.IS_GUTENBERG_PLUGIN`, but just using a `IS_GUTENBERG_PLUGIN` global (like the `SCRIPT_DEBUG` one that was added recently) makes more sense. Web scripts shouldn't artificially introduce a `process.env` that doesn't exist. Also weird that the global isn't replaced in build scripts 🤔 

----

> like we do in [Gutenberg](https://github.com/WordPress/gutenberg/blob/3ad84197cad8737fbf9d37fae521564e02502965/tools/webpack/shared.js#L67) and https://github.com/vitejs/vite/issues/1973.

Tip: when linking to files on GitHub, copy its permalink via the three-dots menu or press Y on the keyboard to get the permalink. This way we can see the file in the version at the time of writing, even when the file is later modified or moved.

Reply:
> Web scripts shouldn't artificially introduce a process.env that doesn't exist

For bundled scripts yes, but packages can ship and support environment variables. process.env variables have always been the standard in the JS community for env variables but lately import.meta is becoming the standard when it comes to ES modules.

Reply:
Personally, I'd love if we can experiment with this https://github.com/WordPress/gutenberg/pull/60087#discussion_r1581023317 and use consistently for all of our env variables. SCRIPT_DEBUG is a runtime variable, so it's different.

That said, I don't want to block this PR more. I'm fine with mentioning the variable in our docs temporarily while we figure this out.

Reply:
I mentioned it in slack, but will mention it here too. There's also usage of `process.env.NODE_ENV` in the codebase (in packages like `blocks`, `components`, `data`).

I don't know if build tools handle that property in a special way or if it also causes the same issues as `IS_GUTENBERG_PLUGIN`. It might be the latter, but it's just that `IS_GUTENBERG_PLUGIN` causes an error first.

Just mentioning it as I'd hate to see effort go into fixing `IS_GUTENBERG_PLUGIN` but then later realize there's still a problem with `NODE_ENV`.